tests/end-to-end/docker-compose.yaml

```diff
@@ -5,16 +5,38 @@ services:
     environment:
       POSTGRES_PASSWORD: test
       POSTGRES_DB: test
+  postgresql2:
+    image: docker.io/postgres:13.1-alpine
+    environment:
+      POSTGRES_PASSWORD: test2
+      POSTGRES_DB: test
+      POSTGRES_USER: postgres2
+    ports:
+      - "5433:5432"
```

Comment:
I don't think this is being used. Especially since you're not connecting to this database on port 5433. You can probably remove this.

Reply:
I think I should keep this and try to run postgres on the container's port 5433, but I wasn't able to do that. I found this - https://stackoverflow.com/questions/37775702/changing-a-postgres-containers-server-port-in-docker-compose but  I don't think it worked. I want to do 5433 and map it to the host's 5433 so that I can test the --restore option without the default, as you mention below.

Reply:
Oh, the answer I linked seems to do what I want for postgres, I'll look for solutions for the other databases too.

Reply:
Okay done for all, let me know if this is better.

Reply:
So I think your `command: -p 5433` is a good fix here, because it means the second Postgres is running on a different port than the first one, so you can test the port overriding feature of your new code. But I still don't think the `ports: - "5433:5432"` is doing anything for your tests, because they're going through the container network rather than the host network.. You don't need to expose the port at all. (Unless I'm sorely mistaken..)

Reply:
Yup, it worked without the ports: thing, thanks!